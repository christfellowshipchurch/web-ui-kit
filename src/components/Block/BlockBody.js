import React from 'react'
import { Parser } from 'html-to-react'

const htmlToReactParser = new Parser()

const BlockBody = ({ className, children }) =>
  <div className={className}>
    {htmlToReactParser.parse(children)}
  </div>

const defaultProps = {}
const propTypes = {}

BlockBody.defaultProps = defaultProps
BlockBody.propTypes = propTypes

export default BlockBody
