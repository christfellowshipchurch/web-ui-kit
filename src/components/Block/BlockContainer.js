import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  contentContainer, contentMedia, contentRight, contentLeft, contentTop, contentBottom
} from '../styles/scss/styles.module.scss'

import Media from '../Media'

const BlockContainer = ({
  layout, imageUrl, imageAlt, videoUrl, ratio, children, className, rounded, media
}) => {
  const layouts = {
    default: contentBottom,
    inverted: contentTop,
    left: contentLeft,
    right: contentRight,
  };

  const mediaItem = media
    ? (
      <div className={contentMedia}>
        <Media {...media} />
      </div>
    )
    : imageUrl || videoUrl
      ? (
        <div className={contentMedia}>
          <Media ratio={ratio} imageUrl={imageUrl} imageAlt={imageAlt} videoUrl={videoUrl} rounded={rounded} />
        </div>
      )
      : null

  return (
    <div className={classNames(contentContainer, className)}>
      {mediaItem}
      <div className={layouts[layout]}>
        {children}
      </div>
    </div>
  );
};

const defaultProps = {
  layout: 'default',
  imageUrl: null,
  imageAlt: null,
  videoUrl: null,
  ratio: '16by9',
  className: '',
  rounded: false,
  media: null
};

const propTypes = {
  layout: PropTypes.string,
  imageUrl: PropTypes.string,
  imageAlt: PropTypes.string,
  videoUrl: PropTypes.string,
  ratio: PropTypes.string,
  className: PropTypes.string,
  rounded: PropTypes.bool,
  media: PropTypes.object
};

BlockContainer.defaultProps = defaultProps;
BlockContainer.propTypes = propTypes;

export default BlockContainer;
