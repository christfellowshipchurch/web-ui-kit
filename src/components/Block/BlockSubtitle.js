import React from 'react'

const BlockSubtitle = ({ className, children, larger = false }) => (
  larger ?
    <h5 className={className}>
      {children}
    </h5> :
    <h6 className={className}>
      {children}
    </h6>
)

export default BlockSubtitle
