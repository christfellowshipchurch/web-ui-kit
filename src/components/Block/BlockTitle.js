import React from 'react'

const BlockTitle = ({ className, children, larger = false }) => (
  larger ?
    <h1 className={className}>
      {children}
    </h1> :
    <h2 className={className}>
      {children}
    </h2>
)

export default BlockTitle;
