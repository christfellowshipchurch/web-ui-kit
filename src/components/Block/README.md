# Content

![alt christ-fellowship-flat-ui](/docs/static/Content.png)

## About
The component has the ability to display a piece of media with a title, content, and calls to action.

## Layouts
There are four different layout configurations available: Default, Inverted, Left, and Right.
* **default** - media on top of content
* **inverted** - media below content
* **left** - content sits to the left of media
* **right** - content sits to the right of media

## Usage
The component is made up of the following components: Wrapper, Title, and Body.
* **ContentContainer:** this is the parent container that handles the configuration of the media and content. You are able to pass it any Html or React Component(s) that you want to appear in the body. It accepts the following properties: layout, imageUrl, imageAlt, videoUrl, and ratio.
* **ContentTitle:** optional styled title for the piece of content
* **ContentBody:** optional styled body wrapper for the piece of content that accepts html and React Component(s)


```jsx
<Content
    layout="default"
    imageUrl="htts://img.url"
    imageAlt="Alt text for my image"
    videoUrl="https://vid.url"
    ratio="1by1"
>
    <Content.Title>
        This is the title on my content
    </Content.Title>

    <Content.Body>
        <h4>Here I can pass in any html that I want</h4>

        I can also just pass in plain text

        <ReactComponent>This is my super cool React Component that I'm passing in too!</ReactComponent>
    </Content.Body>

    <i>*I can also just pass in extra Html to the main component</i>
</Content>
```


## Testing
Flat UI Web using jest snapshots for testing.

Running `yarn test` runs a test and outputs the result. When run at the _root_, it will run tests for the entire project. Running at a specific directory will run tests for that directory.

Running `yarn test -u` will update snapshots. **Only do this if you are positive that snapshots need to be updated**.

## Contributing
Pull requests are welcome. For major changes, please open an issue first to discuss what you would like to change.

Please make sure to update tests as appropriate. Commits without tests are not likely to be taken into consideration.

## Authors and acknowledgment
[Caleb Panza](https://github.com/calebpanza)

## License
[MIT](https://choosealicense.com/licenses/mit/)