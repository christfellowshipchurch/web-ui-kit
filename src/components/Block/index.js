import Wrapper from './BlockContainer'
import Title from './BlockTitle'
import Subtitle from './BlockSubtitle'
import Body from './BlockBody'

const Block = Wrapper
Block.Title = Title
Block.Subtitle = Subtitle
Block.Body = Body

export default Block