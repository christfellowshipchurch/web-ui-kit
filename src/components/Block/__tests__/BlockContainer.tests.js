import React from 'react'
import Block from '..'
import renderer from 'react-test-renderer'

describe('BlockContainer', () => {
    it('renders Block in default layout without media', () => {
        const tree = renderer.create(
            <Block
                ratio="21by9"
                layout="default"
            >
                <Block.Title>Title</Block.Title>
                <Block.Body>Block body.</Block.Body>
            </Block>
        )

        expect(tree).toMatchSnapshot();
    })

    it('renders an image and Block in default layout', () => {
        const tree = renderer.create(
            <Block
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Block.Title>Title</Block.Title>
                <Block.Body>Block body.</Block.Body>
            </Block>
        )

        expect(tree).toMatchSnapshot();
    })

    it('renders a video with background image and Block in default layout', () => {
        const tree = renderer.create(
            <Block
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Block.Title>Title</Block.Title>
                <Block.Body>Block body.</Block.Body>
            </Block>
        )

        expect(tree).toMatchSnapshot();
    })

    it('renders a subtitle in the Block in the default size', () => {
        const tree = renderer.create(
            <Block
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Block.Title>Title</Block.Title>
                <Block.Subtitle>Subtitle</Block.Subtitle>
                <Block.Body>Block body.</Block.Body>
            </Block>
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a larger title and subtitle', () => {
        const tree = renderer.create(
            <Block
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="default"
            >
                <Block.Title larger>Title</Block.Title>
                <Block.Subtitle larger>Subtitle</Block.Subtitle>
                <Block.Body>Block body.</Block.Body>
            </Block>
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a content Block with the inverted layout', () => {
    const tree = renderer.create(
            <Block
                videoUrl="https://video.url.com"
                imageUrl="https://image.url.com"
                imageAlt="Image Alt"
                ratio="21by9"
                layout="inverted"
            >
                <Block.Title larger>Title</Block.Title>
                <Block.Subtitle larger>Subtitle</Block.Subtitle>
                <Block.Body>Block body.</Block.Body>
            </Block>
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a content Block with the right layout', () => {
        const tree = renderer.create(
                <Block
                    videoUrl="https://video.url.com"
                    imageUrl="https://image.url.com"
                    imageAlt="Image Alt"
                    ratio="21by9"
                    layout="right"
                >
                    <Block.Title larger>Title</Block.Title>
                    <Block.Subtitle larger>Subtitle</Block.Subtitle>
                    <Block.Body>Block body.</Block.Body>
                </Block>
            )
    
            expect(tree).toMatchSnapshot()
        })

        it('renders a content Block with the left layout', () => {
            const tree = renderer.create(
                    <Block
                        videoUrl="https://video.url.com"
                        imageUrl="https://image.url.com"
                        imageAlt="Image Alt"
                        ratio="21by9"
                        layout="left"
                    >
                        <Block.Title larger>Title</Block.Title>
                        <Block.Subtitle larger>Subtitle</Block.Subtitle>
                        <Block.Body>Block body.</Block.Body>
                    </Block>
                )
        
                expect(tree).toMatchSnapshot()
            })
})