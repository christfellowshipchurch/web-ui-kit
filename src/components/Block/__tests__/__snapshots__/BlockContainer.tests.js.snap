// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockContainer renders Block in default layout without media 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    />
    <div
      className="col-12 col-md text-center px-3"
    >
      <h2>
        Title
      </h2>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders a content Block with the inverted layout 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-last px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <video
            autoPlay={true}
            className="embed-responsive-item"
            loop={true}
            muted={true}
            playsInline={true}
          >
            <source
              src="https://video.url.com"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-center px-3"
    >
      <h1>
        Title
      </h1>
      <h5>
        Subtitle
      </h5>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders a content Block with the left layout 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-6 order-first order-md-last px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <video
            autoPlay={true}
            className="embed-responsive-item"
            loop={true}
            muted={true}
            playsInline={true}
          >
            <source
              src="https://video.url.com"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-left px-3"
    >
      <h1>
        Title
      </h1>
      <h5>
        Subtitle
      </h5>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders a content Block with the right layout 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-6 order-first px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <video
            autoPlay={true}
            className="embed-responsive-item"
            loop={true}
            muted={true}
            playsInline={true}
          >
            <source
              src="https://video.url.com"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-left px-3"
    >
      <h1>
        Title
      </h1>
      <h5>
        Subtitle
      </h5>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders a larger title and subtitle 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <video
            autoPlay={true}
            className="embed-responsive-item"
            loop={true}
            muted={true}
            playsInline={true}
          >
            <source
              src="https://video.url.com"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-center px-3"
    >
      <h1>
        Title
      </h1>
      <h5>
        Subtitle
      </h5>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders a subtitle in the Block in the default size 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <video
            autoPlay={true}
            className="embed-responsive-item"
            loop={true}
            muted={true}
            playsInline={true}
          >
            <source
              src="https://video.url.com"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-center px-3"
    >
      <h2>
        Title
      </h2>
      <h6>
        Subtitle
      </h6>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders a video with background image and Block in default layout 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <video
            autoPlay={true}
            className="embed-responsive-item"
            loop={true}
            muted={true}
            playsInline={true}
          >
            <source
              src="https://video.url.com"
              type="video/mp4"
            />
          </video>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-center px-3"
    >
      <h2>
        Title
      </h2>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockContainer renders an image and Block in default layout 1`] = `
<div
  className="container-fluid "
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-12 col-md-12 order-first px-3"
    >
      <div>
        <div
          className="embed-responsive embed-responsive-21by9 rounded "
        >
          <div
            className="fill d-flex justify-content-center align-items-center"
          >
            <div
              className="spinner-border text-primary"
              role="status"
            >
              <span
                className="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-md text-center px-3"
    >
      <h2>
        Title
      </h2>
      <div>
        Block body.
      </div>
    </div>
  </div>
</div>
`;
